rest/src/routes.rs

```diff
@@ -267,8 +335,28 @@ impl<N: Network, B: BlockStorage<N>, P: ProgramStorage<N>> Server<N, B, P> {
         ledger: Arc<RwLock<Ledger<N, B, P>>>,
     ) -> Result<impl Reply, Rejection> {
         // Fetch the records using the view key.
-        let records =
-            ledger.read().find_records(&view_key, RecordsFilter::Unspent).or_reject()?.collect::<IndexMap<_, _>>();
+        let records = ledger
+            .read()
+            .find_records(&view_key, RecordsFilter::Unspent)
+            .or_reject()?
+            .map(|(_commitment, record)| record)
+            .collect::<Vec<_>>();
+        // Return the records.
+        Ok(reply::with_status(reply::json(&records), StatusCode::OK))
+    }
+
+    /// Returns the unspent records for the given view key.
+    async fn ciphertexts_unspent(
+        view_key: ViewKey<N>,
+        ledger: Arc<RwLock<Ledger<N, B, P>>>,
+    ) -> Result<impl Reply, Rejection> {
+        // Fetch the records using the view key.
+        let ledger_reader = ledger.read();
```

Comment:
this read access should happen inline (like in the previous methods) otherwise this lock is held longer than necessary